lib/perl/Genome/InstrumentData/AlignmentResult.pm

```diff
@@ -1629,9 +1629,112 @@ sub verify_alignment_data {
 }
 
 sub alignment_bam_file_paths {
+    return glob(shift->output_dir . "/*.bam");
+}
+
+sub recreated_alignment_bam_file_paths {
+    my $self = shift;
+
+    my @bams = glob($self->output_dir . "/*.bam");
+    if (@bams) {
+        return @bams;
+    }
+
+    my $recreated_bam = File::Spec->join($self->output_dir, 'all_sequences.bam');
+    my $merged_bam    = $self->get_merged_bam_to_revivify_per_lane_bam;
+
+    unless ($merged_bam and -s $merged_bam) {
+        die $self->error_message('Failed to get valid merged bam to recreate per lane bam.');
+    }
+
+    my $cmd = Genome::InstrumentData::AlignmentResult::Command::RecreatePerLaneBam->create(
```

Comment:
Before we can safely reconstitute the BAM, we must reallocate enough space.  (This might result in the allocation being moved to a new volume.)